### PR TITLE
Test for Dashboard Controller

### LINKED ISSUE #40: tests: write tests for dashboard controller
tests files are named as `*.spec.ts` files.

Checkout such files for controllers for more reference.

### PR DESCRIPTION
**Pull request by tanay13**
fixes #40

### CODE REVIEW

**Comment by tanay13** [comment]
@YashKumarVerma Please review this . Is this the right way?

File: src/dashboard/dashboard.controller.spec.ts [hunk 1]
```diff
@@ -8,13 +10,60 @@ describe('DashboardController', () => {
...
+          useFactory: () => ({
+            create: jest.fn(() => true),
+            findAll: jest.fn(() => true),
+            findOneById: jest.fn(() => true),
+            findAllByOwner: jest.fn(() => true),
+            findAllByParticipant: jest.fn(() => true),
+            pushUserIntoApplicationParticipantList: jest.fn(() => true),
+            findOneByIdAndSecret: jest.fn(() => true),
+          })
+        }
```
Comment (YashKumarVerma) [inline src/dashboard/dashboard.controller.spec.ts:37]: awesome !

File: src/dashboard/dashboard.controller.spec.ts [hunk 2]
```diff
@@ -8,13 +10,60 @@ describe('DashboardController', () => {
...
+      ],
     }).compile();
 
     controller = module.get<DashboardController>(DashboardController);
+    service: module.get<UserService>(UserService);
   });
 
-  it('should be defined', () => {
+  it('should be defined', async() => {
```
Comment (YashKumarVerma) [inline src/dashboard/dashboard.controller.spec.ts:45]: do we need to add async here ?

File: src/dashboard/dashboard.controller.spec.ts [hunk 3]
```diff
@@ -8,13 +10,60 @@ describe('DashboardController', () => {
...
+    it('should be defined',() => {
+      expect(controller.showData).toBeDefined();
+    });
+  });
+  describe('.showDev()',() => {
+    it('should be defined',() => {
+      expect(controller.showDev).toBeDefined();
+    });
+  });
+
```
Comment (YashKumarVerma) [inline src/dashboard/dashboard.controller.spec.ts:68]: dope !

**Review by YashKumarVerma — APPROVED** [review]
could you remove the yarn.lock file from this PR? I'll push a commit with the updated lock file if needed.

Rest all looks fine. 
Also, can you post the screenshot of tests passing?

**Comment by tanay13** [comment]
testres
@YashKumarVerma

**Comment by tanay13** [comment]
@YashKumarVerma I deleted the yarn.lock file from this PR

File: src/dashboard/dashboard.controller.spec.ts [hunk 4]
```diff
@@ -42,11 +42,11 @@ describe('DashboardController', () => {
...
 
-  it('should be defined', async() => {
+  it('should be defined', () => {
     expect(controller).toBeDefined();
   });
-  describe('.showDashboard()', async() => {
-    it('should be defined', async() => {
+  describe('.showDashboard()', () => {
+    it('should be defined', () => {
       expect(controller.showDashboard).toBeDefined();
```
Comment (YashKumarVerma) [inline src/dashboard/dashboard.controller.spec.ts]: looks nice.

**Review by YashKumarVerma — APPROVED** [review]
i guess you have deleted the yarn.lock file. Sorry if I was not clear the last time. I meant removing it from the PR.

Please don't include any change in the `yarn.lock` file in the PR.

**Comment by tanay13** [comment]
> i guess you have deleted the yarn.lock file. Sorry if I was not clear the last time. I meant removing it from the PR.
> 
> Please don't include any change in the `yarn.lock` file in the PR.

I don't know but i didn't make any changes in yarn.lock file in my previous PR also . Let me check once again

**Comment by tanay13** [comment]
@YashKumarVerma  Is it fine if I create another PR and close this one because yarn.lock file got deleted from this branch so i am creating PR from another branch?

**Comment by tanay13** [comment]
@YashKumarVerma Please check #83

**Comment by YashKumarVerma** [comment]
> @YashKumarVerma Is it fine if I create another PR and close this one because yarn.lock file got deleted from this branch so i am creating PR from another branch?

use `git checkout -- yarn.lock` to restore changes in file / restore file

**Comment by YashKumarVerma** [comment]
Closing as fixed in #84

**Pull request closed without merging**